Automattic/wp-calypso
# A4A: Update site error indicator.

This PR improves the display of site errors in the dashboard by adopting a more DataView-compliant design. The previous approach relied on some hacks, which were not ideal.

**NOTE: This PR does not include the sidebar menu for connection errors. We will handle it in a separate PR.**

| Before | After |
|--------|--------|
| <img width="330" alt="Screenshot 2024-10-24 at 6 33 12 PM" src="https://github.com/user-attachments/assets/d179c254-1bd4-4c00-826b-6f4a5090107f"> | <img width="492" alt="Screenshot 2024-10-24 at 6 33 03 PM" src="https://github.com/user-attachments/assets/f4e16659-cb7b-433b-9336-012bbd58521d"> |

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/1018

## Proposed Changes

* Remove the old implementation when dealing with site errors in the dashboard.
* Add a new indicator in the Site column that, when hovered over, will display the errors. Note that we currently only support one error, but we will later expand that to support more errors, such as plugins. 

## Why are these changes being made?

* This removes hacks, which are not ideal for long-term maintenance.

## Testing Instructions

* Use the A4A live link and go to the `/sites` page.
* Disconnect some Jetpack sites to force an error in the dashboard.
* Confirm that you see the new indicator.

## Pre-merge Checklist
